pkg/crd/init.go

```diff
@@ -537,3 +553,36 @@ func (f *Factory) getReadyCRDs(ctx context.Context) (map[string]*apiext.CustomRe
 
 	return result, nil
 }
+
+func ConvertCRDtoV1Beta1(toConvertCRD runtime.Object) (runtime.Object, error) {
+	if toConvertCRD == nil {
+		return nil, fmt.Errorf("cannot convert empty runtime.Object to object with GVK %v", GVKV1Beta1)
+	}
+	if toConvertCRD.GetObjectKind().GroupVersionKind() != GVK {
+		return nil, fmt.Errorf("GVK for runtime.Object is not %v, which is required for conversion to object with GVK %v", GVK, GVKV1Beta1)
+	}
+
+	unstructuredCRD, ok := toConvertCRD.(*unstructured.Unstructured)
+	if !ok {
+		return nil, fmt.Errorf("could not convert runtime.Object with GVK %v to *unstructured.Unstructured", GVK)
+	}
+	var v1CRD *apiextv1.CustomResourceDefinition
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructuredCRD.UnstructuredContent(), &v1CRD); err != nil {
+		return nil, err
+	}
+	var internalCRD *apiext.CustomResourceDefinition
+	if err := apiextv1.Convert_v1_CustomResourceDefinition_To_apiextensions_CustomResourceDefinition(v1CRD, internalCRD, nil); err != nil {
```

Comment:
Nice! Didn't know it has this util function.

Reply:
Yep!